File: Data Structures & Algorithms/car-fleet/submission-1.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        posWV = []
        for i in range(len(position)):
            posWV.append((position[i], speed[i]))
        posWV.sort(reverse=True)

        times = []
        for pos,speed in posWV:
            times.append((float)(target - pos) / speed)
            while len(times) >= 2 and times[-1] <= times[-2]:
                times.pop()
        return len(times)

File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;
        for(int i = 0; i < nums.size() - 2; i++){
            int j = i + 1;
            int k = nums.size() - 1;
            while(j < k && j < nums.size()){
                int target = nums[i] * -1;
                if(nums[j] + nums[k] == target){
                    res.push_back({nums[i], nums[j], nums[k]});
                    break;
                }
                else if(nums[j] + nums[k] > target) k--;
                else j++;
            }
        }
        return res;
    }
};
